Add --only-selected-files flag for selective Solidity file auditing

sumitvekariya opened the pull request:
- Add new CLI flag to allow selecting specific files
- Implement interactive file selection functionality
- Update process_local to handle selective file auditing
- Fix #5

eugenPtr commented on agent/local.py:
```diff
@@ -28,30 +28,86 @@ def clone_repository(repo_url: str) -> str:
+        all_files: List of all Solidity file paths found
+        
+    Returns:
+        List of selected file paths
+    """
+    if not all_files:
+        logger.warning("No Solidity files found in repository")
+        return []
+        
+    print("\nFound the following Solidity files:")
+    for idx, file_path in enumerate(all_files, 1):
+        print(f"{idx}. {file_path}")
+        
+    while True:
+        try:
+            selection = input("\nEnter the numbers of files to audit (comma-separated, e.g. '1,3,4'): ")
+            if not selection.strip():
+                logger.warning("No files selected, exiting...")
+                return []
+                
+            # Parse and validate selections
+            selected_indices = [int(idx.strip()) for idx in selection.split(",")]
+            selected_files = []
+            
+            for idx in selected_indices:
+                if 1 <= idx <= len(all_files):
+                    selected_files.append(all_files[idx - 1])
+                else:
+                    print(f"Invalid selection {idx}, skipping...")
+            
+            if selected_files:
+                print("\nSelected files:")
+                for file in selected_files:
+                    print(f"- {file}")
+                return selected_files
+            else:
+                print("No valid files selected, please try again")
+                
+        except ValueError:
+            print("Invalid input. Please enter comma-separated numbers")
```
Although this approach works, I want to ask for a sleeker user experience (see the image below)

![image](https://github.com/user-attachments/assets/5256436d-9df9-4e44-8919-6bc0d966bb00)

Below is the code implementing the above selection mechanism.
If you choose to go with this code, you will need to do 2 additional things:

1. Add `import questionary` at the top of this file
2. Add `"questionary>=2.0.1",` to the dependencies array in `pyproject.toml`

```suggestion
def select_files_interactively(all_files: List[str]) -> List[str]:
    """
    Display all .sol files and let the user select which ones to audit
    using a modern interactive checkbox interface.
    
    Args:
        all_files: List of all Solidity file paths found
        
    Returns:
        List of selected file paths
    """
    if not all_files:
        logger.warning("No Solidity files found in repository")
        return []
    
    try:
        # Display interactive checkbox selection
        # Note: questionary uses simple dictionaries for choices, not a Choice class
        selected_files = questionary.checkbox(
            "Select Solidity files to audit:",
            choices=all_files,
            instruction="Use arrow keys to navigate, space to select, enter to confirm"
        ).ask()
        
        if selected_files:
            print("\nSelected files:")
            for file in selected_files:
                print(f"- {file}")
            return selected_files
        else:
            logger.warning("No files selected, exiting...")
            return []
            
    except Exception as e:
        logger.error(f"Error in interactive file selection: {str(e)}")
        # Fallback to simple text-based selection
        print("\nFound the following Solidity files:")
        for idx, file_path in enumerate(all_files, 1):
            print(f"{idx}. {file_path}")
            
        while True:
            try:
                selection = input("\nEnter the numbers of files to audit (comma-separated, e.g. '1,3,4'): ")
                if not selection.strip():
                    logger.warning("No files selected, exiting...")
                    return []
                    
                # Parse and validate selections
                selected_indices = [int(idx.strip()) for idx in selection.split(",")]
                selected_files = []
                
                for idx in selected_indices:
                    if 1 <= idx <= len(all_files):
                        selected_files.append(all_files[idx - 1])
                    else:
                        print(f"Invalid selection {idx}, skipping...")
                
                if selected_files:
                    print("\nSelected files:")
                    for file in selected_files:
                        print(f"- {file}")
                    return selected_files
                else:
                    print("No valid files selected, please try again")
                    
            except ValueError:
                print("Invalid input. Please enter comma-separated numbers")
```

eugenPtr reviewed (changes requested):
Thank you so much for taking on the 2 issues! Super excited to see your interest to contribute. 

I left one request. Please make sure the following runs correctly when you're done updating 

`audit-agent local --repo https://github.com/andreitoma8/learn-solidity-hacks.git --output audit.txt --only-selected-files`

sumitvekariya commented:
> Thank you so much for taking on the 2 issues! Super excited to see your interest to contribute.
> 
> I left one request. Please make sure the following runs correctly when you're done updating
> 
> `audit-agent local --repo https://github.com/andreitoma8/learn-solidity-hacks.git --output audit.txt --only-selected-files`

I have addressed your feedback and ran it locally it works, Thanks

<img width="513" alt="image" src="https://github.com/user-attachments/assets/81bc36a3-f2d2-48c4-b183-13273f9caa71" />

eugenPtr commented:
@sumitvekariya have you joined the [Agent4rena tg group](https://t.me/agent4rena) yet? I would like to give you a shoutout for the features you implemented

sumitvekariya commented:
> @sumitvekariya have you joined the [Agent4rena tg group](https://t.me/agent4rena) yet? I would like to give you a shoutout for the features you implemented
> 

Yeah just joined. Username: `sarkazein7`

Thanks

eugenPtr reviewed (approved):
LGTM!
